diff --git a/net/bluetooth/hci_sync.c b/net/bluetooth/hci_sync.c
index 80eab1d0e..c3fc5d4de 100644
--- a/net/bluetooth/hci_sync.c
+++ b/net/bluetooth/hci_sync.c
@@ -3779,0 +3780,3 @@ static int hci_read_local_name_sync(struct hci_dev *hdev)
+	if (test_bit(HCI_QUIRK_BROKEN_READ_LOCAL_NAME, &hdev->quirks))
+		return 0;
+
@@ -3956,0 +3960,3 @@ static int hci_read_local_ext_features_1_sync(struct hci_dev *hdev)
+	if (test_bit(HCI_QUIRK_BROKEN_LOCAL_EXT_FEATURES, &hdev->quirks))
+		return 0;
+
